mkosi/__init__.py

```diff
@@ -2966,16 +2968,14 @@ def copy_git_files(src: str, dest: str, *, source_file_transfer: SourceFileTrans
 
 
 def install_build_src(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> None:
-    if not do_run_build_script:
+    if (not do_run_build_script or args.build_script is None) and not args.prepare_script:
```

Comment:
This conditional is a bit complicated. We bail out of `do_run_build_script` is false or we don't have a `build_script` unless we do have a `prepare_script`, right? Although I'm not a fan of breaking this up in general, maybe this is clearer
```python
if args.prepare_script is None:
    if not do_run_build_script or args.build_script is None:
        return
```
or maybe just
```suggestion
    if not args.prepare_script is None and (not do_run_build_script or args.build_script is None):
```

Reply:
Actually, we need to worry about the final image here as well. We can't just go copying in a ton of source files there without removing them.

Reply:
Pushed a new version that just mounts the source directory in if it exists. I think mounting is fine in this case. We could go further and add an overlay so that packages have to be installed somewhere out of $SRCDIR but I'm not sure if we want this.